stdlib/public/core/FloatingPointTypes.swift.gyb

```diff
@@ -625,7 +625,20 @@ extension ${Self}: BinaryFloatingPoint {
   /// almost never is.
   @_inlineable // FIXME(sil-serialize-all)
   public var ulp: ${Self} {
-    if !isFinite { return ${Self}.nan }
+%if bits == 32 or bits == 64:
+    guard _fastPath(isFinite) else { return .nan }
+    if _fastPath(isNormal) {
+      let bitPattern_ = bitPattern & ${Self}.infinity.bitPattern
+      return ${Self}(bitPattern: bitPattern_) / 0x1p${SignificandBitCount}
```

Comment:
I would tend to write this as `0x1p-${SignificandBitCount}` because I think of it that way, but there shouldn't be any perf difference, since llvm knows how to optimize a known power-of-two divide into a multiply. Probably worth checking that this optimization happens even at -Onone to be safe.

Reply:
Good point. For greatest certainty, I'll just write it out. It's just as clear to the reader in any case.